src/Propulsion.Kafka/Consumers.fs

```diff
@@ -283,62 +283,150 @@ module Core =
-            /// Prevent batches being consolidated prior to scheduling in order to maximize granularity of consumer offset updates
-            ?maximizeOffsetWriting,
-            ?pipelineStatsInterval, ?maxSubmissionsPerPartition, ?pumpInterval, ?logExternalState, ?idleDelay)=
-        Core.StreamsConsumer.Start<'M,(string*Propulsion.Streams.StreamSpan<_>),'Res>(
-            log, config, Bindings.mapConsumeResult, parseStreamEvents, prepare, handle, maxDop, stats,
-            ?pipelineStatsInterval = pipelineStatsInterval,
-            ?maxSubmissionsPerPartition = maxSubmissionsPerPartition,
-            ?pumpInterval = pumpInterval,
-            ?logExternalState = logExternalState,
-            ?idleDelay = idleDelay,
-            ?maximizeOffsetWriting = maximizeOffsetWriting)
-
-    /// Starts a Kafka Consumer running spans of events per stream through the `handle` function to `maxDop` concurrently
-    /// Processor statistics are accumulated serially into the supplied `stats` buffer
-    /// Processor pumps until `Stop()` is requested.
-    /// Handler `Choice1Of2` result must indicate Index at which next processing will proceed (which can trigger discarding of earlier items on that stream)
-    /// Handler `Choice2Of2` result marks the processing of a stream failed (which will then be offered again for retry purposes on the next cycle)
-    static member Start<'M,'Res>
-        (   log : ILogger, config : KafkaConsumerConfig, parseStreamEvents,
-            handle : StreamName * Streams.StreamSpan<_> -> Async<'Res>, maxDop,
-            stats : Streams.Scheduling.StreamSchedulerStats<OkResult<'Res>,FailResult>,
-            /// Prevent batches being consolidated prior to scheduling in order to maximize granularity of consumer offset updates
-            ?maximizeOffsetWriting,
-            ?pipelineStatsInterval, ?maxSubmissionsPerPartition, ?pumpInterval, ?logExternalState, ?idleDelay, ?maxBatches) =
-        Core.StreamsConsumer.Start<'M,'Res>(
-            log, config, Bindings.mapConsumeResult, parseStreamEvents, handle, maxDop, stats,
-            ?pipelineStatsInterval = pipelineStatsInterval,
-            ?maxSubmissionsPerPartition = maxSubmissionsPerPartition,
-            ?pumpInterval = pumpInterval,
-            ?logExternalState = logExternalState,
-            ?idleDelay = idleDelay,
-            ?maxBatches = maxBatches,
-            ?maximizeOffsetWriting = maximizeOffsetWriting)
+        /// Starts a Kafka Consumer running spans of events per stream through the `handle` function to `maxDop` concurrently
+        /// Processor statistics are accumulated serially into the supplied `stats` buffer
+        /// Processor pumps until `Stop()` is requested.
+        /// Handler `Choice1Of2` result must indicate Index at which next processing will proceed (which can trigger discarding of earlier items on that stream)
```

Comment:
Defining a new DU for this may make the handler code more readable and the effect of the return value more obvious, if the `Choice` combinators are not being used.

```
type ConsumeResult =
   | NextIndex of index : int64
   | Retry
```

Also a bit confused by this comment, as `handle` is not returning a `Choice` but an `Async<'Res>`.

Reply:
this has opened up a whole can of very neglected worms, thanks again!
1. will be revising the comments etc - they obviously got out of sync with any reality
2. this has given me a nudge to remove some further mess in there...